projects/storefrontlib/src/cms-components/user/login/login.component.ts

```diff
@@ -1,33 +1,32 @@
-import { Component } from '@angular/core';
-import { AuthService, User, UserService } from '@spartacus/core';
-import { Observable } from 'rxjs';
-import { filter, map, switchMap } from 'rxjs/operators';
-import { LoginComponentService } from './login.component.service';
+import { ChangeDetectorRef, Component, OnInit, OnDestroy } from '@angular/core';
+import { User, UserService } from '@spartacus/core';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'cx-login',
   templateUrl: './login.component.html',
 })
-export class LoginComponent {
+export class LoginComponent implements OnInit, OnDestroy {
+  userDetails: User = {};
+  private subscription = new Subscription();
+
   constructor(
-    private auth: AuthService,
     private userService: UserService,
-    private loginService: LoginComponentService
+    private cd: ChangeDetectorRef
   ) {}
 
-  get user$(): Observable<User> {
```

Comment:
This `login.component` needs to subscribes `userToken` for `user login`. If `userToken` exists (user login), then we get the `user` data.

something like this:
```
get user$(): Observable<User> {
    return this.auth.getUserToken().pipe(
      switchMap(token => {
        if (token && !!token.access_token && !this.loginService.isLogin) {
          return this.userService.get();
        } else if (token && !token.access_token && this.loginService.isLogin) {
          this.loginService.isLogin = false;
          return of();
        }
      })
    );
  }
```

Reply:
Fixed in #2865